src/ray/gcs/gcs_server/gcs_task_manager.cc

```diff
@@ -124,15 +140,29 @@ absl::optional<int64_t> GcsTaskManager::GcsTaskManagerStorage::GetTaskStatusUpda
              : absl::nullopt;
 }
 
+void GcsTaskManager::GcsTaskManagerStorage::MarkTasksFailed(const JobID &job_id,
```

Comment:
Just to make sure I don't miss this; We already have logic to clean children state when the parent fails right? that's why the relevant code has disappeared.

I remember we had N^2 issues as well there. There's no need to resolve in this case?